### DISCUSSION TITLE
Is it possible to set an expiration date for a Salvo session?

### DISCUSSION THREAD

**Discussion by inadati** [salvo-rs/salvo#915] [Q&A]
Hello,
I would like to know how to set an expiration date for a session in Salvo.
Referring to [this](https://salvo.rs/book/features/session.html),
I tried to set the session expiration time using the expire_in method or set_expiry method
both on the handler side and the middleware side,
but the expiration time was always set to the same time the next day and could not be changed.

This is part of my code, but is there anything inappropriate?
```
    let expire = Duration::minutes(1);
    session.expire_in(expire.to_std().map_err(|_| AppErr::Setup("session expire".to_string()))?);
    depot.set_session(session.clone());
```

**Comment by inadati** [comment]
I was able to solve this problem on my own.
According to [this](https://docs.rs/salvo-session/0.72.2/src/salvo_session/lib.rs.html#362-364),
SessionHandler::builder generates a HandlerBuilder,
and HandlerBuilder allows for more detailed session configuration.
Then, HandlerBuilder::build can generate a SessionHandler.
Therefore, when I wanted to set things like expiration time in session handling,
I needed to use HandlerBuilder.

Below is my middleware code that allows setting a custom expiration time.
```
pub async fn middleware() -> anyhow::Result<SessionHandler<RedisSessionStore>> {
    let expiry = Duration::seconds(30).to_std().map_err(|e| e)?;
    let store = RedisSessionStore::new("redis://127.0.0.1:6379/")?;

    Ok(
        HandlerBuilder::new(
            store,
            b"secretabsecretabsecretabsecretabsecretabsecretabsecretabsecretab",
        )
        .session_ttl(Some(expiry))
        .build()?
    )
}
```

**Reply by josejachuf** [reply]
Hi @inadati 

**session_ttl**

let session_handler = SessionHandler::builder(CookieStore::new(), session_secret.as_bytes())
        .session_ttl(Some(Duration::from_secs(60 * 20)))
        .build()
        .unwrap();
